Parser/pegen/pegen.c

```diff
@@ -38,6 +38,43 @@ byte_offset_to_character_offset(PyObject *line, int col_offset)
     return size;
 }
 
+static int
+tokenizer_error(Parser *p)
+{
+    if (PyErr_Occurred()) {
+        return -1;
+    }
+
+    const char *msg = NULL;
+
+    switch (p->tok->done) {
+        case E_TOKEN:
+            msg = "invalid token";
+            break;
+        case E_EOFS:
+            msg = "EOF while scanning triple-quoted string literal";
+            break;
+        case E_EOLS:
+            msg = "EOL while scanning string literal";
+            break;
+        case E_EOF:
+            msg = "unexpected EOF while parsing";
+            break;
+        case E_IDENTIFIER:
+            msg = "invalid character in identifier";
+            break;
+        default:
+            msg = "unknown parsing error";
+    }
+
+
+    PyErr_Format(PyExc_SyntaxError, msg);
+    // There is no reliable column information for this error
+    PyErr_SyntaxLocationObject(p->tok->filename, p->tok->lineno, 0);
```

Comment:
Is that really true? The old parser does seem to point to the end of the line at least for E_EOFS and E_EOLS (I didn't check all the others).

Reply:
That comment was already there. I'm going to look into this in more detail, as soon as we have fixed #47, which is currently in the way.